spark/sql-13/src/itest/scala/org/elasticsearch/spark/integration/AbstractScalaEsSparkSQL.scala

```diff
@@ -1036,7 +1044,7 @@ class AbstractScalaEsScalaSparkSQL(prefix: String, readMetadata: jl.Boolean, pus
   @Test
   def testDataSourcePushDown12And() {
     val df = esDataSource("pd_and")
-    var filter = df.filter(df("reason").isNotNull.and(df("airport").endsWith("O")))
+    var filter = df.filter(df("reason").isNotNull.and(df("tag").equalTo("jan")))
```

Comment:
Could you help me understand why `df("airport").endsWith("O")` changed to `df("tag").equalTo("jan")` here and on `sql-20/AbstractScalaEsSparkSQL.scala:1104`?

Reply:
The endsWith predicate translates into a wildcard query. Usually we would lowercase the `O` character to make the query string be `*o`. Wildcard queries are term level queries and do not have analyzers applied to them. When run in "strict" mode, the query keeps the character uppercased (`*O`) which does not match the term in lucene. Lowercasing the `O` in the test would lead to Spark filtering the data out from the results (since `"o" != "O"`).

I swapped the endsWith predicate on the And-testing-methods because I wanted to bring back positive matches from the query in all test cases instead of assuming empty results when just strict mode is enabled.

Reply:
more concisely: The bug deals with `strict` mode being enabled, and this test shouldn't be affected by that setting. Removing the feature that does depend on the setting from the test (hopefully) makes it simpler to reason about.

Reply:
Ah, ok, thanks for the explanation. LGTM!

Reply:
Thanks!